cli/cmd/inject.go

```diff
@@ -368,17 +368,19 @@ func (options *proxyConfigOptions) overrideConfigs(configs *cfg.All, overrideAnn
 
 	if options.proxyImage != "" {
 		configs.Proxy.ProxyImage.ImageName = options.proxyImage
-		if options.dockerRegistry != "" {
-			configs.Proxy.ProxyImage.ImageName = registryOverride(configs.Proxy.ProxyImage.ImageName, options.dockerRegistry)
-		}
 		overrideAnnotations[k8s.ProxyImageAnnotation] = configs.Proxy.ProxyImage.ImageName
 	}
 
 	if options.initImage != "" {
 		configs.Proxy.ProxyInitImage.ImageName = options.initImage
-		if options.dockerRegistry != "" {
-			configs.Proxy.ProxyInitImage.ImageName = registryOverride(configs.Proxy.ProxyInitImage.ImageName, options.dockerRegistry)
-		}
+
+		overrideAnnotations[k8s.ProxyInitImageAnnotation] = configs.Proxy.ProxyInitImage.ImageName
+	}
+
+	if options.dockerRegistry != "" {
```

Comment:
Personally, I feel that the `registry` override is the more "general" override and should therefore be applied before the more specific `proxy-image` or `init-image` flags.  That way, the more specific can override the general.  What do you think?

Reply:
The registry should only change the registry part of the full image name, leaving the `proxy-image` or `init-image` values unchanged.  In other words, it could override `gcr.io/foobar` to `my-registry.io/foobar`.  So I don't believe that it needs to be applied before the image overrides.

Reply:
Sorry if I'm really stretching my scenario into something that noone would normally do, but my concern is the following: 
```
linkerd inject --enable-debug-sidecar --registry=my.custom.registry/linkerd-io --init-image=gcr.io/linkerd-io/proxy-init
```
In this case, I my intended result would be for the following to be installed:
```
my.custom.registry/linkerd-io/proxy
my.custom.registry/linkerd-io/debug
gcr.io/linkerd-io/proxy-init
```
but the actual would result in:
```
my.custom.registry/linkerd-io/proxy
my.custom.registry/linkerd-io/debug
my.custom.registry/linkerd-io/proxy-init
```
So in reality, the only way I'd truly be able to achieve my intended setup would be to manually update the override annotations in the post-injection deployment.

Again...I'm probably going overboard with mixing registries, but I could see the possibility.  As an FYI, the registry override for debug sidecar isn't working (PR #3873) addresses that particular issue.

Reply:
Ah, I see what you mean now.  Updated!